.github/workflows/publish-snapshots.yml

```diff
@@ -0,0 +1,34 @@
+name: Publish snapshots to maven
+
+on:
+  push:
+    branches:
+      - main
```

Comment:
Probably `main` is not needed here, since the pull request goes to `2.x` branch

Reply:
Totally!